webui/src/pages/repositories/repository/objects.jsx

```diff
@@ -967,73 +968,7 @@ const TreeContainer = ({
-            return results ? results.sort((a, b) => a.path.localeCompare(b.path)) : results;
-        }
-
-        const changesMap = new Map();
-        const directoryChanges = new Map(); // Store change type for directories
-
-        // Map direct changes and identify affected directories
-        changesData.results.forEach((change) => {
-            // Map change type to what EntryRow expects
-            const mappedType = change.type === 'removed' ? 'removed' : change.type === 'added' ? 'added' : 'changed';
-
-            changesMap.set(change.path, mappedType);
-
-            // If this is a prefix entry from changes, mark it directly with its type
-            if (change.path_type === 'common_prefix') {
-                directoryChanges.set(change.path, mappedType);
-            } else {
-                // For file changes, mark parent directories as changed
-                const pathParts = change.path.split('/');
-                for (let i = 1; i < pathParts.length; i++) {
-                    const dirPath = pathParts.slice(0, i).join('/') + '/';
-                    // Only mark as changed if not already marked with a more specific type
-                    if (!directoryChanges.has(dirPath)) {
-                        directoryChanges.set(dirPath, 'changed');
-                    }
-                }
-            }
-        });
-
-        // Add missing items from changes that aren't in the regular results
-        // This includes deleted files and deleted/changed prefixes
-        const missingItems = changesData.results.filter(
-            (change) => !results.find((result) => result.path === change.path),
-        );
-
-        // Merge regular results with change info
-        const enhancedResults = results.map((entry) => {
-            const directChangeType = changesMap.get(entry.path);
-            if (directChangeType) {
-                return { ...entry, diff_type: directChangeType };
-            }
-
-            // Check if this directory contains changes (either directly or has changed children)
-            const directoryChangeType = directoryChanges.get(entry.path);
-            if (entry.path_type === 'common_prefix' && directoryChangeType) {
-                return { ...entry, diff_type: directoryChangeType };
-            }
-
-            return entry;
-        });
-
-        // Add missing items (deleted files, deleted/changed prefixes) to the results
-        const allResults = [
-            ...enhancedResults,
-            ...missingItems.map((item) => {
-                const mappedType = item.type === 'removed' ? 'removed' : item.type === 'added' ? 'added' : 'changed';
-                return {
-                    ...item,
-                    diff_type: mappedType,
-                };
-            }),
-        ];
-
-        // Sort to maintain proper order
-        return allResults.sort((a, b) => a.path.localeCompare(b.path));
+        return mergeResults(results, changesData, showChangesOnly);
```

Comment:
Nit: call the function directly, it's an expression.
```suggestion
const mergedResults = React.useMemo(() => mergeResults(results, changesData, showChangesOnly), [results, changesData, showChangesOnly]);
```